File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 1/21/2026, 10:35:11 AM
1class Solution {
2public:
3    vector<int> twoSum(vector<int>& nums, int target) {
4        unordered_map<int, int> m;
5
6        for (int i = 0; i < nums.size(); i++) {
7            int complement = target - nums[i];
8
9            if (m.find(complement) != m.end()) {
10                return {m[complement], i};
11            }
12
13            m[nums[i]] = i;
14        }
15
16        return {};
17    }
18};
19

File: 1-TwoSum/1-TwoSum.py
# Last updated: 8/1/2025, 11:41:06 PM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        indexed_arr = []
        # map banaya
        for i,num in enumerate(nums):
            indexed_arr.append((num,i))
        #array sort kari
        indexed_arr.sort()
        #two pointer approach lgaai
        right,left = 0,n-1
        while right<left:
            sum  = indexed_arr[right][0]+indexed_arr[left][0]
            if sum  == target:
                return [indexed_arr[right][1],indexed_arr[left][1]]
            elif sum < target:
                right+=1
            else:
                left-=1
        return []